lib/html/static/js/fileSearch.js

```diff
@@ -0,0 +1,143 @@
+/******************************************************************************
+ * (C) British crown copyright 2012-6 Met Office.
+ *
+ * This file is part of Rose, a framework for scientific suites.
+ *
+ * Rose is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Rose is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Rose. If not, see <http://www.gnu.org/licenses/>.
+ *
+ ******************************************************************************/
+var FileLoader = ( function() {
+    // wrapper to highlight the selected line
+    var mark = ['<span class="highlight">', '</span>'];
+	var previous_search = null;
+
+	var user = null;
+	var suite = null;
+	var path = null;
+	var path_in_tar = null;
+
+
+	var buildUrl = function(base, kwargs) {
+		/* Appends kwargs onto the end of a base url is present. */
+		var url = base;
+		var is_first = true;
+		for (i=0; i<kwargs.length; i++) {
+			// check for js null and python None
+			if (kwargs[i][1] && kwargs[i][1] != 'None') {
+				var c = '&';
+				if (is_first) {
+					c = '?';
+					is_first = false;
+				}
+				url += c + kwargs[i][0] + '=' + kwargs[i][1];
+			}
+		}
+		return url;
+	}
```

Comment:
Would it be easier or nicer to use the `data` argument of `jQuery.ajax` to pass data, instead of building your own URL (which may need escaping, etc)?